components/org.wso2.carbon.security.caas/src/main/java/org/wso2/carbon/security/caas/api/module/UsernamePasswordLoginModule.java

```diff
@@ -102,18 +111,51 @@ public boolean login() throws LoginException {
                     CarbonSecurityErrorMessages.UNSUPPORTED_CALLBACK_EXCEPTION.getDescription(), e);
         } catch (IOException e) {
             throw new CarbonSecurityServerException(CarbonSecurityErrorMessages.CALLBACK_HANDLE_EXCEPTION.getCode(),
-                                                    CarbonSecurityErrorMessages.CALLBACK_HANDLE_EXCEPTION
-                                                            .getDescription(), e);
+                    CarbonSecurityErrorMessages.CALLBACK_HANDLE_EXCEPTION.getDescription(), e);
         }
 
         username = usernameCallback.getName();
         password = passwordCallback.getPassword();
 
-
+        uncommittedUser = validateUserPassword(username, password);
+        success = uncommittedUser != null;
 
         //TODO Add Audit logs CARBON-15870
-        success = true;
-        return true;
+        return success;
+    }
+
+    private User validateUserPassword(String userName, char[] credentials)
+            throws CarbonSecurityServerException, CarbonSecurityAuthenticationException {
+        if (credentials == null) {
```

Comment:
No point of checking the array length. It's a fixed size whether we have values or not.